### DISCUSSION TITLE
[2023-11-13] Team GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8291] [Weekly Demo Day]
* Implemented self-type awareness of the Component Browser: #8268

It reads the type of written local variable from expression updates, and - if it is not overshadowed - propose non-static methods of the type. If the type is non known (the expression update hasn't arrived, or it's a lambda parameter), CB display a list of all non-static methods.

(please ignore connections, they are a known issue).

https://github.com/enso-org/enso/assets/3919101/25efd56d-5aaa-4f10-957d-8ed6a40f3d9b

* Implemented [a fix for sometimes failing tests](https://github.com/enso-org/enso/pull/8277), which cause transient CI failures (PR is almost merged).
* Started working on #8252 - this will fix the issue visible in the video above, where we create connections to local lambda variables.

**Comment by MichaelMauderer** [comment]
* Merged #8241
* Merged #8183
* Finished implementation of  #8279
* Working on #8194

[Peek 2023-11-13 14-15.webm](https://github.com/enso-org/enso/assets/1428930/e04e3d57-61ac-41c3-8d87-929f0394cae9)

**Comment by wdanilo** [comment]
Website progress! :) 

Check it out here (for now, still use Chrome on full-screen): https://enso-org.github.io/website2
We plan to be able to release the website with all fixes till the end of the week. List of fixes done / to be done is [here](https://discord.com/channels/401396655599124480/1171101124636446792/1171101124636446792).


https://github.com/enso-org/enso/assets/1623053/7b4d2226-752c-42d7-a8aa-cdda103ac4fa

**Comment by kazcw** [comment]
This week I've been implementing the new AST (#8236). I'm expect to finish the PR tomorrow.

**Comment by somebody1234** [comment]
- Implemented https://github.com/enso-org/enso/pull/8239
- A bunch of progress on the website
  - Add Google Analytics
    - And corresponding changes on dashboard side
  - Add barebones initial implementation for [blog](https://enso-org.github.io/website2/blog) and [docs](https://enso-org.github.io/website2/docs)
  - Some debugging for performance, some bug fixes

![image](https://github.com/enso-org/enso/assets/4046547/eb09cc87-35f4-4635-ab3d-298cd732ebcf)

![image](https://github.com/enso-org/enso/assets/4046547/22ca5f40-ad65-4bec-a869-ae634e250450)

![image](https://github.com/enso-org/enso/assets/4046547/0b94c540-5590-423e-ac41-f512e9a18735)

**Comment by vitvakatu** [comment]
- Merged https://github.com/enso-org/enso/pull/8227 (it needed an important fix + had an issue with reactivity that took time to debug)
- https://github.com/enso-org/enso/issues/8066 Mostly implemented, but not fully integrated yet. I expect to create a PR tomorrow.

**Comment by mwu-tow** [comment]
* Fixed CI runners issues (disk space cleaning).
* Working on electron-builder / fixing the macOS build.
  * Fixing macOS breaks Windows (and vice versa).
  * MSI installer has issues of its own.

**Comment by Frizi** [comment]
- Finished basic widget tree implementation, added ability to create ports on "virtual elements" in preparation for argument placeholder implementation. Reintroduced the concept of "nesting", so that the top-level arguments can have mouse hover areas. PR is now in process of being merged https://github.com/enso-org/enso/pull/8253

https://github.com/enso-org/enso/assets/919491/fd65de7d-fed9-4fa1-bca4-6fe22c34637a

**Comment by JaroslavTulach** [comment]
- recording of presentation about [Enso at GeeCON Prague](https://www.youtube.com/watch?v=8KhrWjICPe8)
- _inline evaluation_ successfully reviewed - #8148
- the _new IR caching_ is finally speeding Enso initialization up - #8207
- upgrade to GraalVM with [Truffle Unchained](https://youtu.be/8KhrWjICPe8?si=ugaq4xuzqBxj7sXg&t=2555) is progressing
